Task 6

enginire4ek opened the pull request:
- Updated dependency versions
 - Added navigation from the pin page to the author page
 - Added functionality to share pin in messengers
 - Small changes in Dagger BusinessModule.kt
 - Update PinsRepository.kt
 - Added a new page "Creators", which contains a list of creators from HomeFragment
 - Added scrolling on pages containing a list of pins
 - Added transition animations on the pins page
 - Added new layouts for landscape orientation

enginire4ek committed:
Task 6
 - Updated dependency versions
 - Added navigation from the pin page to the author page
 - Added functionality to share pin in messengers
 - Small changes in Dagger BusinessModule.kt
 - Update PinsRepository.kt
 - Added a new page "Creators", which contains a list of creators from HomeFragment
 - Added scrolling on pages containing a list of pins
 - Added transition animations on the pins page
 - Added new layouts for landscape orientation

enginire4ek committed:
Task 6
 - Added new use case
 - Some linter fixes

enginire4ek committed:
Task 6
 - Linter fixes (import-ordering)

enginire4ek committed:
Task 6
 - Linter fixes (import-ordering)

102Rudy commented on domain/src/main/java/com/example/pinterbest/domain/repositories/AuthRepository.kt:
```diff
@@ -4,8 +4,8 @@ import com.example.pinterbest.domain.common.ResponseWithCookie
 import com.example.pinterbest.domain.entities.UserSignUp
-import java.lang.Exception
 import kotlinx.coroutines.flow.Flow
+import java.lang.Exception
```
kotlin has its own Exception class, so you can safely delete the line `import java.lang.Exception`

enginire4ek replied:
Deleted

102Rudy commented on presentation/src/main/java/com/example/pinterbest/presentation/ActualPinFragment.kt:
```diff
@@ -69,20 +99,19 @@ class ActualPinFragment : Fragment() {
-                    }
+            viewLifecycleOwner
+        ) { loading ->
+            when (loading) {
+                true -> {
+                    binding.actualPinScreen.visibility = View.GONE
+                    binding.emptyView.visibility = View.GONE
+                    binding.progressBar.visibility = View.VISIBLE
+                }
+                false -> {
+                    binding.progressBar.visibility = View.GONE
                 }
```
nit: looks a little bit strange, why not to use simple `if/else` structure here?

enginire4ek replied:
Fixed here https://github.com/enginire4ek/Pinterbest/pull/13/commits/7e75b3b148c4925e3a8f02f2ffc3c27aaf25c304

102Rudy commented on presentation/src/main/java/com/example/pinterbest/presentation/ActualPinFragment.kt:
```diff
@@ -69,20 +99,19 @@ class ActualPinFragment : Fragment() {
+        ) { loading ->
+            when (loading) {
+                true -> {
+                    binding.actualPinScreen.visibility = View.GONE
+                    binding.emptyView.visibility = View.GONE
```
don't you need to make `actualPinScreen` and `emptyView` visible when `loading` is false?

enginire4ek replied:
Fixed here https://github.com/enginire4ek/Pinterbest/pull/13/commits/7e75b3b148c4925e3a8f02f2ffc3c27aaf25c304

102Rudy commented on presentation/src/main/java/com/example/pinterbest/presentation/CreatorsFragment.kt:
```diff
@@ -33,5 +52,54 @@ class CreatorsFragment : Fragment() {
+        viewModel.state.observe(
+            viewLifecycleOwner
+        ) { loading ->
+            when (loading) {
+                true -> binding.progressBar.visibility = View.VISIBLE
+                false -> binding.progressBar.visibility = View.GONE
+            }
```
again, I would suggest to replace with `binding.progressBar.visibility = if (loading) View.VISIBLE else View.GONE`

enginire4ek replied:
Fixed here https://github.com/enginire4ek/Pinterbest/pull/13/commits/7e75b3b148c4925e3a8f02f2ffc3c27aaf25c304

102Rudy commented on presentation/src/main/java/com/example/pinterbest/presentation/CreatorsFragment.kt:
```diff
@@ -33,5 +52,54 @@ class CreatorsFragment : Fragment() {
+        )
+    }
+
+    private fun showError(it: String?) {
```
should be non optional

enginire4ek replied:
Fixed here https://github.com/enginire4ek/Pinterbest/pull/13/commits/7e75b3b148c4925e3a8f02f2ffc3c27aaf25c304

102Rudy commented on presentation/src/main/java/com/example/pinterbest/presentation/adapters/CreatorsAdapter.kt:
```diff
@@ -0,0 +1,55 @@
+
+    fun updateList(profilesList: List<ProfileViewData>) {
+        profiles = profilesList
+        notifyDataSetChanged()
```
`notifyDataSetChanged` is considered as bad practice. It's much better to rewrite using DiffUtils

102Rudy commented on presentation/src/main/java/com/example/pinterbest/presentation/adapters/CreatorsAdapter.kt:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    companion object {
+        const val BASE_URL_IMAGES = "https://pinterbest-bucket.s3.eu-central-1.amazonaws.com/"
```
this URL **must** be already included in `profile.avatarLink` field. It's not responsibility of adapters to decide which base url to use for images. For example you can concat avatarLink with BASE_URL_IMAGES in Repository

enginire4ek replied:
Fixed here https://github.com/enginire4ek/Pinterbest/pull/13/commits/7e75b3b148c4925e3a8f02f2ffc3c27aaf25c304

102Rudy commented on presentation/src/main/java/com/example/pinterbest/presentation/viewmodels/CreatorsViewModel.kt:
```diff
@@ -0,0 +1,45 @@
+    private val getPinsCreatorsUseCase: GetPinsCreatorsUseCase
+) : ViewModel() {
+    private val _state = MutableLiveData(false)
+    val state: LiveData<Boolean> = _state
```
`state` of what? Each name should be self documented. Ideally, anybody who sees your code should clearly understand what exactly does variable/class/method without further code exploration. Let's add a little bit of context into the name

enginire4ek replied:
Renamed

102Rudy commented on presentation/src/main/java/com/example/pinterbest/presentation/viewmodels/ProfileViewModel.kt:
```diff
@@ -45,6 +47,23 @@ class ProfileViewModel @Inject constructor(
+            getProfileDetailsByIdUseCase(userId).collect { result ->
+                when (result) {
+                    is Result.Success -> {
+                        _profile.value = result.data
+                        _state.value = false
```
I do not remember exactly, but please double check: probably you need to use `postValue` here instead of just `value = ...`

enginire4ek replied:
No need

enginire4ek committed:
Task 6
 - Fixed comments to the code
 - Pin image display fixed, image is not cropped now

enginire4ek committed:
Task 6
 - Linter fixes
